### PR TITLE
waybar: make module a freeform module, remove warnings

### PR DESCRIPTION
**Pull request by berbiche**
### Description

The `style` option now also accepts a path instead of a text configuration.

Keeping up with new Waybar options is annoying, so make the module a freeform module.

The `modules` option could not be removed because `mkRemovedOptionModule` does not work on submodules with `attrsOf` or `listOf`.

The logic to generate warnings for modules and everything was removed. I don't want to maintain the code that generates these warnings anymore.

_ps: sorry for the single commit, I can split it up if required._

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/programs/waybar.nix [hunk 1]
```diff
@@ -241,12 +197,17 @@ in {
     systemd.enable = mkEnableOption "Waybar systemd integration";
 
     style = mkOption {
-      type = nullOr str;
+      type = nullOr (oneOf [ path str ]);
```
Comment (rycee) [inline modules/programs/waybar.nix:200]: When there are only two elements I think either is a bit prettier:
```suggestion
      type = nullOr (either path str);
```

**Review by rycee — COMMENTED** [review]
Looks fine to me and definitely simpler. About the `modules` removal, it may be possible to add the assertion manually? The `mkRemovedOptionModule` simply evaluates to an assignment to `config.assertions`.

Alternatively, could add an entry in the news/release notes saying that the option will be removed in the 22.05 release? Then can remove it completely in December/January under the assumption that people read news/release notes 😉

**Comment by berbiche** [comment]
I added an assertion that will disable the `modules` option for 22.05 as well as a test case.

The test case will need `version.nix` to be updated before uncommenting the gatekeeping code.

**Comment by rycee** [comment]
Nice, merged to master.

**Comment by berbiche** [comment]
I'll add a release note entry in a different PR.

**Pull request merged**